app/models/user.rb

```diff
@@ -344,6 +345,11 @@ def must_have_assignments_if_not_admin
       end
     end
 
+    def clear_assignments_without_roles
+      empty_assignments = self.assignments.select{ |a| a if a.role.blank? }
+      self.assignments.delete empty_assignments
```

Comment:
You could refactor this as `assignments.delete(assignments.select(&:no_role?))`.